### ISSUE TITLE
In elm-pages v3, Query string cannot be obtained in the init function of Shared.elm.

### ISSUE THREAD

**Issue by ababup1192** [dillonkearns/elm-pages#361]
Sample code.

```
init :
    Pages.Flags.Flags
    ->
        Maybe
            { path :
                { path : Path
                , query : Maybe String
                , fragment : Maybe String
                }
            , metadata : route
            , pageUrl : Maybe PageUrl
            }
    -> ( Model, Effect Msg )
init flags maybePagePath =
    let
        _ =
            Debug.log (Debug.toString maybePagePath) ""
    in
    ...
```
<img width="991" alt="スクリーンショット 2023-01-15 15 22 43" src="https://user-images.githubusercontent.com/1853727/212783093-6d46eebe-4763-456d-9457-a9121239f5ff.png">

Elm debugger shows the query string.

<img width="276" alt="スクリーンショット 2023-01-15 15 21 16" src="https://user-images.githubusercontent.com/1853727/212783136-db2674f6-90cc-46fb-a2ab-14ea76df1838.png">

**Comment by ymtszw** [comment]
I believe it is available at runtime. That is why Elm debugger shows the query string.

Your screenshot must be a build log from terminal. There, elm-pages cannot assume anything about query parameters at build-time. Correct me if I'm wrong @dillonkearns

**Comment by dillonkearns** [comment]
That’s correct @ymtszw, and in general servers do not get access to the query string, they are only available on the client side (so even on server rendered routes, the query string is not sent over as part of the server request so it is invisible to the server). So yes, this is expected 👍

**Comment by dillonkearns** [comment]
That is also the reason that it is a Maybe.
